lib/Parse/ParseStmt.cpp

```diff
@@ -1421,8 +1462,17 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
                                   StringRef &BindingKindStr) {
   ParserStatus Status;
 
-  // Parse a leading #available condition if present.
-  if (Tok.is(tok::pound_available)) {
+  // Diagnose !#available as being a wrong spelling of #unavailable.
+  if (Tok.getText() == "!" && peekToken().is(tok::pound_available)) {
+    diagnose(Tok, diag::false_available_is_called_unavailable)
+      .fixItReplace(SourceRange(Tok.getLoc(), 
+                      peekToken().getLoc()), "#unavailable");
```

Comment:
Nit: formatting is off. `peekToken().getLoc()` should probably go on the earlier line.